pkg/container/container.go

```diff
@@ -244,7 +244,10 @@ func (c Container) runtimeConfig() *dockercontainer.Config {
 		}
 	}
 	for p := range c.containerInfo.HostConfig.PortBindings {
-		config.ExposedPorts[p] = struct{}{}
+		// config.ExposedPorts isn't reliably in sync with HostConfig.PortBindings, so we can't assign directly to map without checking
+		if _, ok := config.ExposedPorts[p]; ok {
+			config.ExposedPorts[p] = struct{}{}
```

Comment:
The whole point of this code is to empty out map entries, so we should be able to safely skip that action if the map is `nil`

Reply:
I'm not sure that it is... Couldn't we just assign ExposedPorts to a new map if it's nil instead? If we keep part of the config as `nil` we might run into issues in other parts of the code that expects it to be a map.
I think the point of this code is to explicitly add the portbindings to the exposed ports (but with no explicit mapping config), but I may be wrong here.
Since the docker daemon on Synology doesn't return any exposed ports, it's somewhat natural that it doesn't require them in the create container payload. It might break container creation on non-synology docker though...